lib/GlobalStylesheets.js

```diff
@@ -30,13 +30,66 @@ function addStyle(css){
 
 function createStylesheet(stylesheet) {
   var style = assign({}, stylesheet.style);
+  var hoverStyles = {};
+  var activeStyles = {};
+  var focusStyles = {};
+
+  Object.keys(style).forEach(function(key) {
+    if (key.indexOf('hover') === 0) {
+      // Strip the key of the pseudo selector prefix and transform first char
+      // to lowercase to ensure correct markup generation and autoprefixing
+      var prop = key.substr(5);
+      hoverStyles[prop.charAt(0).toLowerCase() + prop.slice(1)] = style[key];
+      // Remove the key from main style entry
+      delete style[key];
+    }
+    if (key.indexOf('focus') === 0) {
+      var prop = key.substr(5);
+      focusStyles[prop.charAt(0).toLowerCase() + prop.slice(1)] = style[key];
+      delete style[key];
+    }
+    if (key.indexOf('active') === 0) {
+      var prop = key.substr(6);
+      activeStyles[prop.charAt(0).toLowerCase() + prop.slice(1)] = style[key];
+      delete style[key];
+    }
+  });
+
   autoprefix(style);
   var stylesheetText = (
     '\n.' + PREFIX + stylesheet.id + ' {\n' +
       CSSPropertyOperations.createMarkupForStyles(style) +
       '\n}\n'
   );
 
+  // Generate CSS for any pseudo selector props
+  if (Object.keys(hoverStyles).length > 0) {
```

Comment:
Since `hoverStyles` is null unless a key is added, you can simply check `if (hoverStyles) {` and save an array allocation